ports/webassembly/proxy_js.js

```diff
@@ -244,6 +246,10 @@ function proxy_convert_mp_to_js_obj_jsside(value) {
     } else {
         // obj
         const id = Module.getValue(value + 4, "i32");
+        const existing_obj = globalThis.proxy_js_dict[id];
+        if (existing_obj !== undefined) {
+            return existing_obj.deref();
```

Comment:
watch out `.deref()` doesn't guarantee the proxy is returned and, when that's the case, this branch would fail expectations.

An easier way is to reach out for that proxy instead and keep going if not there as it means nobody used that before but now it's needed again, it should be re-registered and re-assigned.

In code words:

```js
const existing_proxy = globalThis.proxy_js_dict[id]?.deref();
if (existing_proxy) return existing_proxy;
```

Another hint would be to make `proxy_js_dict` a `proxy_js_map` instead as JS is not optimized to `delete` entries from dictionaries but this might be premature optimization too, yet the check would be a bit more elegant via `proxy_js_map.has(id)` and `proxy_js_map.get(id).deref()` but it's OK as it is now, after all.